libcnb-test/CHANGELOG.md

```diff
@@ -2,4 +2,4 @@
 
 ## [Unreleased]
 
-- Initial release
+- Initial release ([#277](https://github.com/Malax/libcnb.rs/pull/277)).
```

Comment:
Passing side comment: I find the requirement that we have links to the current PR in the changelog a less-than-ideal requirement. It means I have to open the PR, grab the URL, update the changelog, force push. I get that it's a minor papercut, but it's a bit annoying nevertheless. 

My preference has been to lessen this burden somewhat by reducing/removing formatting. Having to have the PR number in there twice is room to mess up, having a specific format is room to mess up. I know it's not as aesthetically pleasing but that's the motivation behind my "minimal" formatting  in https://github.com/heroku/heroku-buildpack-ruby/blob/main/CHANGELOG.md

Super-side-note: I would LOVE as a hackday project if we could write a github bot that after a PR is merged with a changelog entry, it auto generates a PR that adds a link back to the original PR. (Or another alternative). The idea that every PR requires at a minimum of 2 pushes makes me feel slightly weary.

That is to say: I prefer if we choose to use simpler formatting for changelog links. I would prefer it even more if we didn't have to manually update the links (as a pie in the sky dream).

Reply:
>  I know it's not as aesthetically pleasing

For me, the desire to have PR links is not related to aesthetics, but usability. 

As an end user of a library, it's extremely inconvenient to have to manually track down the PR in which a change occurred, to read more about it. For example, when a breaking change is mentioned in the changelog and I'd like to see what exactly changed, and the diff for the examples, to make it easier to translate into the changes needed in my usage of a library.

Manual strategies for going from changelog entry to PR include either searching for the PR using keywords from the changelog entry (which doesn't always work since the PR title might not match the changelog entry sufficiently), or else git blame on the changelog, which means a few more clicks and then having to find the same entry in the changelog again after the page reload (and hoping that nothing modified that changelog entry since the original PR, eg to fix a changelog typo).

>  It means I have to open the PR, grab the URL, update the changelog, force push. I get that it's a minor papercut, but it's a bit annoying nevertheless.

I agree it can be annoying.

One strategy I use to avoid the need to double-push (to add the changelog entry), is to look at the last repo issue/PR number (whilst waiting for `cargo test` etc to run locally before pushing), add one to that number and then double check when the PR opens that it was indeed that number. For repos with low levels of activity this strategy works pretty well.